applications/serial_lte_modem/src/main.c

```diff
@@ -172,31 +227,44 @@ int slm_indicate(void)
 
 static void indicate_stop(void)
 {
-#if (CONFIG_SLM_INDICATE_PIN >= 0)
+#if (INDICATE_PIN_IS_ENABLED)
 	if (gpio_pin_set(gpio_dev, CONFIG_SLM_INDICATE_PIN, 0) != 0) {
 		LOG_WRN("GPIO_0 set error");
 	}
 	LOG_DBG("Stop indicating");
 #endif
 }
 
-static void gpio_cb_func(const struct device *dev, struct gpio_callback *gpio_callback,
-			 uint32_t pins)
+#if POWER_PIN_IS_ENABLED
+
+static void power_pin_callback_enable_poweroff(const struct device *dev,
+					       struct gpio_callback *gpio_callback, uint32_t)
 {
-	int err;
+	static K_WORK_DELAYABLE_DEFINE(work, (k_work_handler_t)enable_poweroff_interrupt);
 
-	if ((BIT(CONFIG_SLM_WAKEUP_PIN) & pins) == 0) {
-		return;
-	}
+	LOG_DBG("Enabling the poweroff interrupt shortly...");
+	gpio_remove_callback(dev, gpio_callback);
+
+	/* Enable the poweroff interrupt after a small delay
```

Comment:
I'm not sure what you mean exactly. But I'm tempted to say that it doesn't matter, because the work is scheduled and does not rely on any pin value. Maybe even 1ms would suffice.

Reply:
Ah, ok, I thought that this would work in a way that when in idle, it would wakeup with level triggered interrupt, after which it would then enable the edge triggering and would hit when the button was released for this same press.

Reply:
If I understand you correctly, that's the whole purpose of the interrupt configured when waking up. To prevent the power off interrupt from firing up right away, another one is set up to trigger and configure the power off interrupt when the pin is released during the wake-up operation.

Reply:
Yeah, my mistake was thinking that there would be timeout after receiving wakeup, but the timeout is after the rising-edge from wakeup is processed and intends to avoid hitting the same edge again.